### PR TITLE
copy logic for warehouse to stage and truncate stage

### PR DESCRIPTION
**Pull request by chdietz**
pushing to keep the code safe, need to resolve merger conflicts.
Have one validation for table count to figure our for job test.

### CODE REVIEW

File: migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/repository/impl/JdbcWarehouseToStageRepository.java [hunk 1]
```diff
@@ -81,4 +81,17 @@ public JdbcWarehouseToStageRepository(final NamedParameterJdbcTemplate warehouse
     }
 
 
+    @Override
+    public void truncateStaging(Iterable<String> stagingTruncateStatements) {
+
+        for ( String truncate : stagingTruncateStatements ) {
+            reportingJdbcTemplate.getJdbcOperations().execute(truncate);
+        }
+
```
Comment (malaffoon) [inline migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/repository/impl/JdbcWarehouseToStageRepository.java]: That, sir, is a lot of whitespace.

File: migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/repository/impl/JdbcWarehouseToStageRepository.java [hunk 2]
```diff
@@ -81,4 +81,17 @@ public JdbcWarehouseToStageRepository(final NamedParameterJdbcTemplate warehouse
     }
 
 
+    @Override
+    public void truncateStaging(Iterable<String> stagingTruncateStatements) {
```
Comment (malaffoon) [inline migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/repository/impl/JdbcWarehouseToStageRepository.java]: nit: `final` for param and for-loop variable

File: migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/step/SqlCopyExecutionStep.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                                final List<SqlCopyStatements> sqlCopyStatements) {
+        this.sqlCopyStatements = sqlCopyStatements;
+        this.supportedContents = supportedContents;
+        this.repository = repository;
+    }
+
+    @Override
+    public RepeatStatus execute(final StepContribution contribution, final ChunkContext chunkContext) throws Exception {
+
+        // hack for now
```
Comment (malaffoon) [inline migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/step/SqlCopyExecutionStep.java]: This isn't going to be merged to develop, right?

File: migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/repository/SqlCopyWarehouseToStagingRepository.java [hunk 4]
```diff
@@ -2,7 +2,11 @@
...
 
 import org.opentestsystem.rdw.ingest.migrate.reporting.MigrateBatch;
+import org.opentestsystem.rdw.ingest.migrate.reporting.step.SqlCopyStatements;
+
+import java.util.List;
 
 public interface SqlCopyWarehouseToStagingRepository {
-    void execute(String warehouseReadSql, String stagingInsertSql, int migrateId, MigrateBatch migrateBatch);
+
+    void execute(List<SqlCopyStatements> copyStatementsList, int migrateId, MigrateBatch migrateBatch);
```
Comment (agorina) [inline migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/repository/SqlCopyWarehouseToStagingRepository.java]: I do not think you need a migrateId. MigrateBatch has it.

File: migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/step/WarehouseCheck.java [hunk 5]
```diff
@@ -15,28 +16,28 @@
...
      * @throws UnsupportedOperationException to signal the step has failed.
      */
     @Override
     public RepeatStatus execute(final StepContribution contribution, final ChunkContext chunkContext) throws Exception {
-
-        final List<Map<String, Object>> reportingSubjectClaimScores =
-                warehouseToStageRepository.findReportingSubjectClaimScores();
-
         if (checkSubject() && checkAsmtTypes() && checkSubjectClaimScores()) {
+            warehouseToStageRepository.truncateStaging(warehouseToStagingSqlConfiguration.getTruncateStageList());
```
Comment (agorina) [inline migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/step/WarehouseCheck.java]: I was hoping that 'truncate' would be a separate step that we can call as the first step and the last step.

File: migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/repository/impl/JdbcWarehouseToStageRepository.java (hunk 1 above)
Comment (chdietz) [inline migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/repository/impl/JdbcWarehouseToStageRepository.java]: yeah, need to do another pass on this.

File: migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/step/SqlCopyExecutionStep.java (hunk 3 above)
Comment (chdietz) [inline migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/step/SqlCopyExecutionStep.java]: Correct.  I did the PR so you all can review.  I wanted to see WH-stage-reporting running end to end.  Need to change this and decide where we want to put truncate, deal with merge conflicts and one assertion error in one of the IT tests.

File: migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/step/WarehouseCheck.java (hunk 5 above)
Comment (chdietz) [inline migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/step/WarehouseCheck.java]: I was wondering about that, can easily make it it's own step.  Maybe after the check step I slipped it in now?

Comment (agorina) [inline migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/step/WarehouseCheck.java]: Sounds good, thank you.

Comment (chdietz) [inline migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/step/WarehouseCheck.java]: Will address this in next round of changes.

**Pull request merged**